base/src/main/java/org/aya/distill/CoreDistiller.java

```diff
@@ -259,25 +259,29 @@ private ImmutableSeq<Arg<Term>> visibleArgsOf(Callable call) {
       linkRef(term.ref(), FIELD_CALL));
   }
 
-  public @NotNull Doc pat(@NotNull Pat pat, Outer outer) {
+  public @NotNull Doc pat(@NotNull Arg<Pat> pat, @NotNull Outer outer) {
+    return pat(pat.term(), pat.explicit(), outer);
+  }
+
+  public @NotNull Doc pat(@NotNull Pat pat, boolean licit, Outer outer) {
     return switch (pat) {
       case Pat.Meta meta -> {
         var sol = meta.solution().get();
-        yield sol != null ? pat(sol, outer) : Doc.bracedUnless(linkDef(meta.fakeBind()), meta.explicit());
+        yield sol != null ? pat(sol, licit, outer) : Doc.bracedUnless(linkDef(meta.fakeBind()), licit);
       }
-      case Pat.Bind bind -> Doc.bracedUnless(linkDef(bind.bind()), bind.explicit());
+      case Pat.Bind bind -> Doc.bracedUnless(linkDef(bind.bind()), licit);
       case Pat.Ctor ctor -> {
-        var ctorDoc = visitCalls(ctor.ref(), CON_CALL, ctor.params().view().map(Pat::toArg), outer,
+        var ctorDoc = visitCalls(ctor.ref(), CON_CALL, Arg.mapSeq(ctor.params().view(), Pat::toTerm), outer,
           options.map.get(AyaDistillerOptions.Key.ShowImplicitPats));
-        yield ctorDoc(outer, ctor.explicit(), ctorDoc, ctor.params().isEmpty());
+        yield ctorDoc(outer, licit, ctorDoc, ctor.params().isEmpty());
       }
-      case Pat.Absurd absurd -> Doc.bracedUnless(Doc.styled(KEYWORD, "()"), absurd.explicit());
-      case Pat.Tuple tuple -> Doc.licit(tuple.explicit(),
-        Doc.commaList(tuple.pats().view().map(sub -> pat(sub, Outer.Free))));
+      case Pat.Absurd absurd -> Doc.bracedUnless(Doc.styled(KEYWORD, "()"), licit);
+      case Pat.Tuple tuple -> Doc.licit(licit,
+        Doc.commaList(tuple.pats().view().map(sub -> pat(sub, true, Outer.Free))));   // TODO: Always true?
```

Comment:
Not always true I believe

Reply:
`({a}, b)` is possible?

Reply:
I handled this case. I think this is yet impossible from the concrete but possible from elaboration.